### PR TITLE
Maintain Ironic Endpoint with Keepalived

### PR DESCRIPTION
**Pull request by kashifest**
Co-authored-by: Xenwar <anwar.hassen@est.tech>

This PR ensures that the ironic-endpoint in minikube is maintained using keepalived (for pivoting). To ensure that we add:

- Add ironic-endpoint bridge as the interface both in minikube and target cluster contronplane
- Change the BMO kustomization path to ironic-keepalived-config. This ensures BMO deployment also creates the keepalived container to maintain the VIP.

It also performs few necessary changes in controlplane CR accordingly. Namely:

- Adds Calico CNI
- Adds ironicendpoint in netplan so that the networking survives reboot
- Add keepalived and related configurations.

### CODE REVIEW

**Comment by kashifest** [comment]
/hold 
This PR should wait for https://github.com/metal3-io/baremetal-operator/pull/402 to go in first.

File: crs/v1alpha2/controlplane_ubuntu.yaml [hunk 1]
```diff
@@ -42,22 +42,30 @@ spec:
     - mv /tmp/akeys /home/ubuntu/.ssh/authorized_keys
     - chown ubuntu:ubuntu /home/ubuntu/.ssh/authorized_keys
     - apt update -y
+    - netplan apply
     - >-
-      apt install net-tools gcc linux-headers-$(uname -r) keepalived
+      apt install net-tools gcc linux-headers-$(uname -r) bridge-utils
```
Comment (Xenwar) [inline crs/v1alpha2/controlplane_ubuntu.yaml]: I do not think we need these.
gcc
linux-headers-$(uname -r)

Comment (kashifest) [reply crs/v1alpha2/controlplane_ubuntu.yaml]: This PR is not adding those packages so I have not checked whether we can live without those

File: crs/v1alpha2/cluster.yaml [hunk 2]
```diff
@@ -20,5 +20,5 @@ kind: BareMetalCluster
 metadata:
   name: ${CLUSTER_NAME}
 spec:
- apiEndpoint: http://192.168.111.249:6443
+ apiEndpoint: http://${CLUSTER_APIENDPOINT_IP}:6443
```
Comment (stbenjam) [inline crs/v1alpha2/cluster.yaml]: Hm, this URL doesn't look like it's constructed in a way that's compatible with IPv6. You can see an example of how we do this with `$CLUSTER_URL_HOST`, basically IPv6 addresses in the host part of a URL need the surrounded by square brackets.

Comment (kashifest) [reply crs/v1alpha2/cluster.yaml]: Thanks for pointing this out. I have changed the code accordingly

Comment (kashifest) [reply crs/v1alpha2/cluster.yaml]: Thanks for point this out. I have changed the code accordingly

File: crs/v1alpha2/controlplane_ubuntu.yaml (hunk 1 above)
Comment (Xenwar) [reply crs/v1alpha2/controlplane_ubuntu.yaml]: I agree.

**Pull request merged**